tests/test_group_by.cc

```diff
@@ -589,4 +590,52 @@ TEST(test_groupby, test_group_by_any_all) {
   EXPECT_FALSE(grouped.apply(greater_than_max_count).any());
 }
 
+TEST(test_groupby, test_group_by_with_vector) {
+
+  const auto fib = fibonacci(20);
+
+  std::vector<std::string> strings;
+  for (const auto &x : fib) {
+    strings.push_back(std::to_string(x));
+  }
+
+  const auto grouped_with_strings =
+      group_by(fib, number_of_digits).with(strings);
+  EXPECT_GT(grouped_with_strings.size(), 0);
+  for (const auto &group : grouped_with_strings) {
+    const std::vector<double> &vector_of_doubles = group.second.first;
+    const std::vector<std::string> &vector_of_strings = group.second.second;
+    for (std::size_t i = 0; i < vector_of_doubles.size(); ++i) {
+      EXPECT_EQ(std::to_string(vector_of_doubles[i]), vector_of_strings[i]);
+    }
+  }
+}
+
+TEST(test_groupby, test_group_by_with_map) {
+
+  const auto fib = fibonacci(20);
+
+  auto to_strings = [](const std::vector<double> &xs) {
+    std::vector<std::string> strings;
+    for (const auto &x : xs) {
+      strings.push_back(std::to_string(x));
+    }
+    return strings;
+  };
+
+  const auto grouped = group_by(fib, number_of_digits);
+
+  const auto as_strings = grouped.apply(to_strings);
```

Comment:
nit, maybe move the helper `to_strings` just before where it is used, and change `to_strings` to `double_to_strings`?